astropy/time/core.py

```diff
@@ -1255,6 +1256,33 @@ def to(self, *args, **kwargs):
                           u.day).to(*args, **kwargs)
 
 
+def _regexify_subfmts(subfmts):
+    """
+    Iterate through each of the sub-formats and try substituting simple
+    regular expressions for the strptime codes for year, month, day-of-month,
+    hour, minute, second.  If no % characters remain then turn the final string
+    into a compiled regex.  This assumes time formats do not have a % in them.
+    """
```

Comment:
I'd add a single sentence of *why* we do this -- both to speed up parsing of strings and to allow mixed formats where `strptime` does not quite work well enough.